src/modules/uORB/uORBDevices_posix.cpp

```diff
@@ -421,31 +422,17 @@ uORB::DeviceNode::appears_updated(SubscriberData *sd)
 			break;
 		}
 
-		/*
-		 * If the interval timer is still running, the topic should not
-		 * appear updated, even though at this point we know that it has.
-		 * We have previously been through here, so the subscriber
-		 * must have collected the update we reported, otherwise
-		 * update_reported would still be true.
-		 */
-		if (!hrt_called(&sd->update_call)) {
+		// If we have not yet reached the deadline, then assume that we can ignore any
+		// newly received data.
+		if (sd->last_update + sd->update_interval > hrt_absolute_time()) {
 			break;
 		}
 
 		/*
-		 * Make sure that we don't consider the topic to be updated again
-		 * until the interval has passed once more by restarting the interval
-		 * timer and thereby re-scheduling a poll notification at that time.
-		 */
-		hrt_call_after(&sd->update_call,
-			       sd->update_interval,
-			       &uORB::DeviceNode::update_deferred_trampoline,
-			       (void *)this);
```

Comment:
@jywilson @julianoes There was latency introduced by the HRT queue processing. Jim made this change while debugging the latency.